### PR TITLE
Add state filter tests and handle readonly database

### PR DESCRIPTION
**Pull request by 183amir**
fix: more error handling
    
    Handle cases when job_manager.logs_dir is not relative to logs.
    Also output valid states when state callback fails.

fix: when the database is read-only
    
    It can happen that the database could be read-only but running commands like gridtk list should still work.

test: add tests for state-based filtering
    
    Adding some tests to make sure state-based filtering of jobs works

----
📚 Documentation preview 📚: https://gridtk--8.org.readthedocs.build/en/8/

### CODE REVIEW

**Review by Yannick-Dayer — COMMENTED** [review]
Looks good.

Can you just confirm the output path comment (that it is doing what is expected).

File: src/gridtk/cli.py [hunk 1]
```diff
@@ -384,6 +385,48 @@ def resubmit(
...
+            job_ids=job_ids, states=states, names=names, dependents=dependents
+        )
+        table = defaultdict(list)
+        for job in jobs:
+            table["job-id"].append(job.id)
+            table["slurm-id"].append(job.grid_id)
+            table["nodes"].append(job.nodes)
+            table["state"].append(f"{job.state} ({job.exit_code})")
+            table["job-name"].append(job.name)
+            output = job_manager.logs_dir / job.output_files[0].resolve()
```
[the remark this answers is not in the document]
Comment (183amir) [reply src/gridtk/cli.py:415]: `.resolve()` does not output absolute paths, it resolves symbolic links only.

[the remark this answers is not in the document]
Comment (183amir) [reply src/gridtk/cli.py:415]: sorry looks like I `.resolve()` returns absolute paths :)

**Review by Yannick-Dayer — APPROVED** [review]
Seems much more logical to me like this.

File: src/gridtk/cli.py [hunk 2]
```diff
@@ -41,6 +41,97 @@ def get_command(self, ctx, cmd_name):
...
+            help="Selects jobs based on their states separated by comma. Possible values are "
+            + ", ".join([f"{v} ({k})" for k, v in JOB_STATES_MAPPING.items()])
+            + " and ALL.",
+            callback=states_callback,
+        )(function)
+        function = click.option(
+            "-j",
+            "--jobs",
+            "job_ids",
+            help="Selects only these job ids, separated by comma.",  # TODO: explain range notation
```
[the remark this answers is not in the document]
Comment (183amir) [reply src/gridtk/cli.py:122]: could you please commit that here? Then, I can merge it.

**Review by Yannick-Dayer — APPROVED** [review]

**Pull request merged**